### DISCUSSION TITLE
Editor / viewport behavior when selecting other files

### DISCUSSION THREAD

**Discussion by Pawper** [beakerbrowser/beaker#1661] [General]
Say you're working on a hyperdrive and editing an html file using the editor.
![image](https://user-images.githubusercontent.com/22269691/82741797-b9c2b300-9d0a-11ea-8a03-4154c1d5c9b6.png)

If you select a link css file to do your styling it opens in the editor AND the viewport as well (for clarity I am using "viewport" instead of "window" since that may refer to the entire application window).
![image](https://user-images.githubusercontent.com/22269691/82741784-9435a980-9d0a-11ea-8fd3-f0be5920553f.png)

This means you cannot preview your changes unless you pop out the editor, which means you don't get the live reload.
![image](https://user-images.githubusercontent.com/22269691/82741812-f5f61380-9d0a-11ea-9f6f-eca52fecaca2.png)

I'm not convinced this is a bug. I think it's reasonable that you may open a file aside from editing and expect to view it in the viewport without the editing context. For example, if you are just exploring someone else's hyperdrive using the Explore Files pane. Editor is neither needed nor wanted.
![image](https://user-images.githubusercontent.com/22269691/82741771-5d5f9380-9d0a-11ea-92c0-eadb07fcb901.png)

However, there is a problem here - I want live reload of the html file while editing a css file. Ideally I wouldn't have to pop out the editor either. Ideas are:
- For the file explorer pane specifically within the editor, do not open in the viewport unless it's a markdown or html file. So if I select a CSS file it should not change the viewport.
![image](https://user-images.githubusercontent.com/22269691/82741750-386b2080-9d0a-11ea-9333-1aa2b15d90af.png)
- An option to right-click a file and view in viewport. Opens the file in the viewport but not the editor. Should only work if the editor isn't popped out, since that could get confusing with multiple tabs or application windows.

Other ideas?

**Comment by pfrazee** [comment]
Yeah I do think some way to control this would be nice. I'll consider your suggestions and look for a solution here.
